bpo-37488 : Document a warning for datetime.utcnow() and utcfromtimestamp() (#15773)
python/cpython: Doc/library/datetime.rst

interpretation [#]_.

A naive object does not contain enough information to unambiguously locate

> itself relative to other date/time objects.  Native objects are no longer

Reviewer: Was this an intentional change? It looks like you kept in the beginnings of an explanation of the problem, but also the problematic old text.

This is actually a bit complicated to reword, because the sentiment behind it has changed, and we may need to hedge a bit. Let me think about how best to communicate the new intent....

   :class:`.datetime` object.  An aware current UTC datetime can be obtained by
   calling ``datetime.now(timezone.utc)``.  See also :meth:`now`.

   .. warning::

>       Because naive datetimes are treated by many functions as local times,

Reviewer 2: This should be a bit more specific. Also, code comments or Sphinx roles should be used when referencing objects (such as classes, methods, or functions). In this case, code comments would be more appropriate since this is already within the section for ``datetime``. 

```suggestion
      Because naive ``datetime`` objects are treated by many ``datetime`` methods as local times,
```

This line will need to be split off into the next one since it will exceed the reST maximum width of 80 chars (94 chars after the suggestion).

>       it is recommended that you create a timezone-aware datetime

Reviewer 2: Phrasing improvement, it's preferable to avoid the usage of pronouns such as "you" when possible within technical writing, and I think "timezone specific" makes a bit more sense than "timezone-aware". 
```suggestion
      it is recommended to create a timezone specific ``datetime`` object
```

Author: Thanks for the review. I consulted @pganssle at the sprints and he says we should keep to aware here. I hope I have addressed the other suggestions.

Reviewer 2: @nanjekyejoannah:
> Thanks for the review. I consulted @pganssle at the sprints and he says we should keep to aware here. I hope I have addressed the other suggestions.

Sounds good to me. If Paul prefers using "timezone-aware" over "timezone specific", I have no objections.

   except the latter formula always supports the full years range: between
   :const:`MINYEAR` and :const:`MAXYEAR` inclusive.

   .. warning::

>       Because naive datetimes are treated by many functions as local times.

Reviewer 3: Can you change this wording to be of a similar sentence structure as in the earlier section (for `datetime.utcnow()`)?

>       To get an aware :class:`.datetime` object, use :meth:`fromtimestamp`.

Reviewer 2: The Sphinx role here for ``datetime`` can be removed here since this section is already within the ``datetime`` class. The main purpose of the roles is to provide links to other sections, so there's not much of a purpose in a role that links to a section it is already contained within.

```suggestion
      To get a timezone specific datetime object, use :meth:`fromtimestamp`.
```
